remote_install_checkbox_snaps/remote_install_checkbox_snaps.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+"""
+The checkbox snap also needs to install a runtime snap, which is specific to
+the OS version of the image is running. We try to select the correct runtime
+based on the track of checkbox snap being installed. If the runtime snap is
+specified then we install that one instead.
+"""
+import argparse
+import logging
+import shlex
+import subprocess
+import sys
+from dataclasses import dataclass
+from typing import List, Optional
+
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclass
+class CheckboxOptions:
+    checkbox_snap: str
+    checkbox_channel: str
+    checkbox_track: str
+    checkbox_args: str
+    checkbox_runtime: Optional[str] = None
+
+
+class SSHClient:
+    def __init__(self, host: str, username: str):
+        self.host: str = host
+        self.username: str = username
+
+    def execute_command(self, command: str):
+        logging.info("Executing command on %s: %s", self.host, command)
+        try:
+            ssh_command = (
+                f"ssh {self.username}@{self.host} {shlex.quote(command)}"
+            )
+            subprocess.check_output(ssh_command, shell=True)
+        except subprocess.SubprocessError as e:
+            sys.exit(f"SSH error executing command on {self.host}: {e}")
+
+
+class RemoteCheckboxInstaller:
+    def __init__(
+        self,
+        ssh_client: SSHClient,
+        options: CheckboxOptions,
+    ):
+        self.ssh_client: SSHClient = ssh_client
+        self.options: CheckboxOptions = options
+
+    def install_snaps(self) -> None:
+        """Install the checkbox snaps on the remote host"""
+
+        runtime_install_cmd = self.get_checkbox_runtime_install_cmd()
+        logging.info(
+            "Running '%s' on %s", runtime_install_cmd, self.ssh_client.host
+        )
+        self.ssh_client.execute_command(runtime_install_cmd)
+
+        checkbox_install_cmd = self.get_checkbox_install_cmd()
+        logging.info(
+            "Running '%s' on %s", checkbox_install_cmd, self.ssh_client.host
+        )
+        self.ssh_client.execute_command(checkbox_install_cmd)
+
+    def get_checkbox_runtime(self) -> str:
+        """Get the name of the checkbox runtime snap for the given track"""
+        if self.options.checkbox_runtime:
+            return self.options.checkbox_runtime
+        track_digits: str = "".join(
+            filter(str.isdigit, self.options.checkbox_track)
+        )[:2]
+        return f"checkbox{track_digits}"
+
+    def get_checkbox_runtime_install_cmd(self) -> str:
+        """Get the install command for the checkbox runtime snap"""
+        return (
+            f"sudo snap install {self.get_checkbox_runtime()} "
+            f"--channel=latest/{self.options.checkbox_channel}"
```

Comment:
In [`run-sru.sh`](https://github.com/canonical/hwcert-jenkins-jobs/blob/6d3bc41c557a8c8b110857f1c85ecc99029d49db/jobs/sru/run-sru.sh#L224C13-L224C66) the channel is always `beta`. Here, it is assumed that the channel will be provided (and, from the examples, it doesn't look like it's always going to be `beta`). What's the logic that determines what the channel should be? Does it need to be captured in this tool or externally?

Reply:
I think for deb SRUs, that's probably always beta. However it's not universally true that we always want beta. For instance, the checkbox team may want to install edge or candidate in some of their tests. The main example is snaps, where we probably want beta (hence the whole reason for needing something like this), but for oem checkbox snaps, we don't always have the option of installing it from beta as the OEMs control the brand stores. Many of those checkbox snaps need to come from stable or edge, and this is already captured in the `{checkbox-channel}` value for those jobs.